Sources/Sentry/SentrySwizzleWrapper.m

```diff
@@ -0,0 +1,78 @@
+#import "SentrySwizzleWrapper.h"
+#import "SentrySwizzle.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentrySwizzleWrapper ()
+
+#if SENTRY_HAS_UIKIT
+@property (nonatomic, strong)
+    NSMutableDictionary<NSString *, SentrySwizzleSendActionCallback> *swizzleSendActionCallbacks;
+#endif
+
+@end
+
+@implementation SentrySwizzleWrapper
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+#if SENTRY_HAS_UIKIT
+        self.swizzleSendActionCallbacks = [NSMutableDictionary new];
+#endif
+    }
+    return self;
+}
+
+#if SENTRY_HAS_UIKIT
+- (void)swizzleSendAction:(SentrySwizzleSendActionCallback)callback forKey:(NSString *)key
+{
+    // We need to make a copy of the block to avoid ARC of autoreleasing it.
+    self.swizzleSendActionCallbacks[key] = [callback copy];
+
+    if (self.swizzleSendActionCallbacks.count != 1) {
+        return;
+    }
+
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    __block SentrySwizzleWrapper *_self = self;
```

Comment:
h: The swizzling method will have a reference to the first ever `SentrySwizzleWrapper` that swizzled things. Lets say, for some reason, the user calls `SentrySDK.close()` and later starts it again, this will reset `SentryDependencyContainer` and creates another instance of `SentrySwizzleWrapper`.

I know this is very unlikely, but also very possible.

Reply:
Excellent catch 👏